provider/core/src/data_provider.rs

```diff
@@ -110,11 +110,11 @@ impl<'data, T: IsCovariant<'data>> ErasedDestructor<'data> for T {}
 /// use a trait object to handle that.
 pub(crate) type ErasedCart<'data> = Rc<dyn ErasedDestructor<'data> + 'data>;
 
-pub(crate) enum DataPayloadInner<'data, M>
+pub(crate) enum DataPayloadInner<M>
 where
     M: DataMarker,
 {
-    RcStruct(Yoke<M::Yokeable, ErasedCart<'data>>),
+    RcStruct(Yoke<M::Yokeable, ErasedCart<'static>>),
```

Comment:
I didn't remove the lifetime from ErasedCart since I'll probably do that when we remove RcStruct (and move ErasedStruct to yoke)